### PR TITLE
fix batch_norm cuda kernel

### PR DESCRIPTION
**Pull request by JZZ-NOTE**
### PR types

Others
### PR changes

Others
### Describe

当 reserve_space 为 null 时，为其分配空间

### CODE REVIEW

**Comment by phlrain** [comment]
batch norm已经被迁移到新的算子库中了，需要升级, paddle/phi/kernels/gpu/batch_norm_kernel.cu 文件

**Review by phlrain — APPROVED** [review]
